### ISSUE TITLE
Different way of importing seems to change code semantics

### ISSUE THREAD

**Issue by CodingSpiderFox** [Automattic/mongoose#13116]
### Prerequisites

- [X] I have written a descriptive issue title

### Mongoose version

6.10.0

### Node.js version

18.14.1

### MongoDB version

4.2

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

elementaryOS 7 (Ubunt 22.04)

### Issue

I am still having trouble wrapping my head around JavaScript and TypeScript...

Why do I get an error?

```
FAIL   api-e2e  apps/api-e2e/src/api/api.spec.ts (14.929 s)
  Post twice, then get
    ✕ should return two messages (14119 ms)

  ● Post twice, then get › should return two messages

    TypeError: Cannot read properties of undefined (reading 'useDb')

      25 |
      26 |     await mongoose.connect(DB_URI);
    > 27 |     const apiDb = mongoose.connection.useDb(DEFAULT_DB_NAME);
```

after changing the way I import mongoose in this commit: https://github.com/CodingSpiderFox/nx-neste2e-mcve/commit/80386ee977af9f8404d525b296a10114941aa984

Disclaimer:
a) After spending hours solving this issue: https://github.com/Automattic/mongoose/issues/13111 I am almost falling asleep now :D

**Comment by Uzlopak** [comment]
What happens if you set esModuleInterop to true in tsconfig?

**Comment by CodingSpiderFox** [comment]
It already is true.
This is the `tsconfig.json`

``
{
  "extends": "../../tsconfig.base.json",
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.spec.json"
    }
  ],
  "compilerOptions": {
    "esModuleInterop": true,
    "resolveJsonModule": true
  }
}
```

`tsconfig.base.json`:

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "rootDir": ".",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "importHelpers": true,
    "target": "es2015",
    "module": "esnext",
    "lib": ["es2017", "dom"],
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "baseUrl": ".",
    "paths": {}
  },
  "exclude": ["node_modules", "tmp"]
}
```

**Comment by vkarpov15** [comment]
We'll look into it, but we don't recommend doing `import * as mongoose from 'mongoose'` in general. Mongoose doesn't export a POJO, it exports an instance of the `Mongoose` class, and `import * as mongoose from 'mongoose'` coerces the export into a POJO. You can do `import { model } from 'mongoose'`, and `import mongoose from 'mongoose'` is probably the best approach.

**Comment by vkarpov15** [comment]
We took a closer look and this is due to TypeScript using Node's ESM named import semantics for `import * as mongoose from 'mongoose'`, which doesn't do well with computed properties like `connection`. We've tried myriad ways to export connection with Node ESM named import semantics, haven't been able to find a way without breaking changes. See #12256.

We advise using `import mongoose from 'mongoose'` over `import * as mongoose from 'mongoose'`.

But if you must use `import * as mongoose from 'mongoose'`, we instead recommend destructuring the return value from `mongoose.connect()` to get the connection, rather than importing it:

```javascript
// Instead of this:
import * as mongoose from 'mongoose';
await mongoose.connect(DB_URI);
const apiDb = mongoose.connection.useDb(DEFAULT_DB_NAME);

// Do this:
import * as mongoose from 'mongoose';
const { connection } = await mongoose.connect(DB_URI);
const apiDb = connection.useDb(DEFAULT_DB_NAME);
```
